### PR TITLE
issue #87 - fixing issue with rejected channel promise

### PR DESCRIPTION
**Pull request by bfitzpat**
Signed-off-by: Brian Fitzpatrick <bfitzpat@redhat.com>

### CODE REVIEW

**Comment by bfitzpat** [comment]
Not sure about this one - the issue does go away, but I'm not convinced we're actually disposing of the output channel under the covers.

File: src/extension.ts [hunk 1]
```diff
@@ -230,11 +236,13 @@ export function hideStatusLine() {
...
 function removeOutputChannel( channelName: string) {
 	let podOutputChannel;
-	if (outputChannelMap.has(channelName)) {
+	if (outputChannelMap !== undefined) {
 		podOutputChannel = outputChannelMap.get(channelName);
 		if (podOutputChannel !== undefined) {
-			podOutputChannel.dispose();
 			outputChannelMap.delete(channelName);
+			// concerned that this doesn't actually delete the resources associated with the output channel, 
+			// but dispose causes other problems
```
Comment (apupier) [inline src/extension.ts]: what are the other problems with dispose?

Comment (bfitzpat) [reply src/extension.ts]: We were disposing the channel before (podOutputChannel.dispose(); on extension.ts line 236 or so) and for some reason there seems to be a thread somewhere still trying to update it in the background AFTER it's been disposed, which was what was creating the "channel has been closed" error I think. Dispose is what SHOULD be called it seems, but doesn't look like it's behaving quite as expected.

Comment (apupier) [reply src/extension.ts]: it means that we are not disposing the pod output channel at all now, isn't it? If yes, sounds like a potential memory leak.

Comment (bfitzpat) [reply src/extension.ts]: We dispose of all output channels when the extension is deactivated. https://github.com/camel-tooling/vscode-camelk/blob/bce6fa9d91e9518cdb00c57372a6695da01d81ed/src/extension.ts#L473

But yes, other than that we are not disposing of them.

Comment (apupier) [reply src/extension.ts]: the extensions not deactivated in a classic usage.
it means that we are trading errors in log against a memory leak. I prefer the errors in log.

Comment (bfitzpat) [reply src/extension.ts]: Ok. As the errors in the debug log don't affect the user, we'll just let it be then. Will close this PR

**Comment by bfitzpat** [comment]
Closing this PR per the conversation

**Pull request closed without merging**